From 858e6f13f364110410e47c08dbdbdc85b4450ebd Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Stefan=20D=C3=B6singer?= <stefan@codeweavers.com>
Date: Wed, 11 Mar 2026 10:11:46 +0300
Subject: [PATCH] d3d8: Build a valid box in d3d8_vertexbuffer_Lock.

This causes a WARN otherwise. Wined3d ignores the box validation error
in this case, so it is relatively harmless, but still misleading.

(cherry picked from commit da9f8090bfcdd6cdf607d6c984688a1d8709af85)
---
 dlls/d3d8/buffer.c | 5 ++---
 1 file changed, 2 insertions(+), 3 deletions(-)

diff --git a/dlls/d3d8/buffer.c b/dlls/d3d8/buffer.c
index f49f6f9f847..bc061e48419 100644
--- a/dlls/d3d8/buffer.c
+++ b/dlls/d3d8/buffer.c
@@ -186,7 +186,7 @@ static HRESULT WINAPI d3d8_vertexbuffer_Lock(IDirect3DVertexBuffer8 *iface, UINT
     struct d3d8_vertexbuffer *buffer = impl_from_IDirect3DVertexBuffer8(iface);
     struct wined3d_resource *wined3d_resource;
     struct wined3d_map_desc wined3d_map_desc;
-    struct wined3d_box wined3d_box = {0};
+    struct wined3d_box wined3d_box;
     HRESULT hr;
 
     TRACE("iface %p, offset %u, size %u, data %p, flags %#lx.\n",
@@ -200,8 +200,7 @@ static HRESULT WINAPI d3d8_vertexbuffer_Lock(IDirect3DVertexBuffer8 *iface, UINT
     if (flags & D3DLOCK_DISCARD)
         buffer->discarded = true;
 
-    wined3d_box.left = offset;
-    wined3d_box.right = offset + size;
+    wined3d_box_set(&wined3d_box, offset, 0, offset + size, 1, 0, 1);
     wined3d_resource = wined3d_buffer_get_resource(buffer->wined3d_buffer);
     hr = wined3d_resource_map(wined3d_resource, 0, &wined3d_map_desc, &wined3d_box,
             wined3dmapflags_from_d3dmapflags(flags, buffer->usage));
